packages/notte-browser/src/notte_browser/controller.py

```
@@ -179,6 +199,118 @@ async def execute_interaction_action(self, window: BrowserWindow, action: Intera
+                                selectors = new_locator_node.computed_attributes.selectors
+
+                                if selectors is not None:
+                                    if selectors.in_shadow_root:
+                                        if self.verbose:
+                                            logger.info(
+                                                f"🔍 Resolving shadow root selectors for {new_locator_node.id} ({new_locator_node.text})"
+                                            )
+                                        selectors = selectors_through_shadow_dom(new_locator_node)
+
+                                    logger.info("Getting new locator")
+                                    locator = await locate_element(window.page, selectors)
```

Pylint (3.3.7)

[refactor] Too many nested blocks (7/5)

(R1702)